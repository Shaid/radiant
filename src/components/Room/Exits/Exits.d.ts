export interface IState {
}

export interface IProps {
  action: Function
  exits: Array<IExit>
}

export interface IExit {
  id: number
  description: Array<string>
  destination: number
}

export interface IDefaultProps {
}

export type PropsWithDefaults = IProps & IDefaultProps
