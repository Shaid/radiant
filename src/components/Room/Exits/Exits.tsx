import React from 'react'
import styled from 'theme'

import Action from 'components/Action'
import {
  IExit, IProps, IDefaultProps, IState
} from './Exits.d'

const Exits = styled.nav`
  display: flex;
  flex-flow: column nowrap;
  font-size: 1rem;
  align-items: flex-start;
`

const Exit = styled.div`
  padding: 0.5rem 0;
  display: inline-block;
`

export default class extends React.PureComponent<IProps, IState> {
  static defaultProps: Partial<IDefaultProps> = {}

  render() {
    const { action, exits } = this.props

    const exitActions = exits.map((exit: IExit) => (
      <Exit key={exit.id}>
        <Action onClick={() => action(exit.destination)}>
          {exit.description}
        </Action>
      </Exit>
    ))

    return (
      <Exits>
        {exitActions}
      </Exits>
    )
  }
}
