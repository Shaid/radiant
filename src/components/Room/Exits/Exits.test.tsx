import React from 'react'
import { mount, shallow } from 'enzyme'

import Exits from '.'
import { IExit } from './Exits.d'

const exits: Array<IExit> = [
  {
    id: 0,
    description: ['A door'],
    destination: 1
  },
  {
    id: 1,
    description: ['A hole'],
    destination: 0
  }
]

describe('<Exits />', () => {
  it('renders', () => {
    const action = jest.fn()

    const wrapper = shallow(<Exits action={action} exits={exits} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('triggers its action when clicked', () => {
    const action = jest.fn()

    const wrapper = mount(<Exits action={action} exits={exits} />)
    wrapper.find('button').first().simulate('click')
    expect(action).toHaveBeenCalled()
  })
})
