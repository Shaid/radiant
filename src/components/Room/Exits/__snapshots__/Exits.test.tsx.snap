// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Exits /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    action={[Function]}
    exits={
      Array [
        Object {
          "description": Array [
            "A door",
          ],
          "destination": 1,
          "id": 0,
        },
        Object {
          "description": Array [
            "A hole",
          ],
          "destination": 0,
          "id": 1,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Exits__Exit>
          <_default
            onClick={[Function]}
          >
            A door
          </_default>
        </Exits__Exit>,
        <Exits__Exit>
          <_default
            onClick={[Function]}
          >
            A hole
          </_default>
        </Exits__Exit>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": <_default
            onClick={[Function]}
          >
            A door
          </_default>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "A door",
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "A door",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": <_default
            onClick={[Function]}
          >
            A hole
          </_default>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "A hole",
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "A hole",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Exits__Exit>
            <_default
              onClick={[Function]}
            >
              A door
            </_default>
          </Exits__Exit>,
          <Exits__Exit>
            <_default
              onClick={[Function]}
            >
              A hole
            </_default>
          </Exits__Exit>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": <_default
              onClick={[Function]}
            >
              A door
            </_default>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "A door",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "A door",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": <_default
              onClick={[Function]}
            >
              A hole
            </_default>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "A hole",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "A hole",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
